tasks/shared_library_posting_to_queue/shared_lib_posting_to_status_queue.py

```diff
@@ -0,0 +1,98 @@
+    if request_time is not None:
+        new_data['request_time'] = request_time
+    if completion_time is not None:
+        new_data['completion_time'] = completion_time
+    if archive_destination is not None:
+        new_data['archive_destination'] = archive_destination
+
+    post_entry_to_queue('orca_recoveryjob',
+                        new_data,
+                        request_method, db_queue_url, max_retries, retry_sleep_secs)
+
+def post_status_for_file_to_queue(job_id: str, granule_id: str, filename: str, key_path: Optional[str],
+                                  restore_destination: Optional[str],
+                                  status_id: ORCA_STATUS, error_message: Optional[str],
+                                  request_time: Optional[str], last_update: str,
+                                  completion_time: Optional[str],
+                                  request_method: RequestMethod,
+                                  db_queue_url: str,
+                                  max_retries: int, retry_sleep_secs: float):
+    new_data = {'job_id': job_id,
+                'granule_id': granule_id,
+                'filename': filename}
+    if key_path is not None:
+        new_data['key_path'] = key_path
+    if restore_destination is not None:
+        new_data['restore_destination'] = restore_destination
+    if status_id is not None:
+        new_data['status_id'] = status_id
+    if error_message is not None:
+        new_data['error_message'] = error_message
+    if request_time is not None:
+        new_data['request_time'] = request_time
+    if last_update is not None:
+        new_data['last_update'] = last_update
+    if completion_time is not None:
+        new_data['completion_time'] = completion_time
+
+    post_entry_to_queue('orca_recoverfile',
+                        new_data,
+                        request_method,
+                        db_queue_url, max_retries, retry_sleep_secs)
+
+
+sqs = boto3.client('sqs')
+
+def post_entry_to_queue(table_name: str, new_data: Dict[str, Any], request_method: RequestMethod, db_queue_url: str,
+                        max_retries: int, retry_sleep_secs: float):
+    body = json.dumps(new_data, indent=4)
+    for attempt in range(1, max_retries + 1):
+        try:
+            sqs.send_message(
+                QueueUrl=db_queue_url,
+                MessageDeduplicationId=table_name + request_method.value + body,
+                MessageGroupId='request_files',
+                MessageAttributes={
+                    'RequestMethod': {
+                        'DataType': 'String',
+                        'StringValue': request_method.value
+                    },
+                    'TableName': {
+                        'DataType': 'String',
+                        'StringValue': table_name
+                    }
+                },
+                MessageBody=body
+            )
+            return
+        except Exception as e:
```

Comment:
Dead code in this 'except' block. You can just remove the entire try/except since we just want errors to bubble up.